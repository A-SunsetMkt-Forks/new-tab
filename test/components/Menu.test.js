'use strict';

const Menu = require('../../src/components/Menu.html');

describe('Menu component', () => {
  it('renders correctly', () => {
    const target = document.createElement('div');
    new Menu({ target });
    expect(target.innerHTML).toMatchSnapshot();
  });

  it('opens the settings page', () => {
    expect.assertions(1);
    const target = document.createElement('div');
    new Menu({ target });
    const link = target.querySelector('#o');
    const spy = jest.spyOn(chrome.runtime, 'openOptionsPage', 'get');
    link.click();
    expect(spy).toHaveBeenCalled();
    spy.mockRestore();
  });
});
