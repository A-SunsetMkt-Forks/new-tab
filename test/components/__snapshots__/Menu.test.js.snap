// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu component renders correctly 1`] = `
"<div id=\\"menu\\"><div id=\\"icon\\">â˜°</div>

			  <div id=\\"dropdown\\"><a href=\\"chrome-search://local-ntp/local-ntp.html\\" class=\\"menu-item\\">
			      Open Default Tab
			    </a>
			    <a href=\\"chrome://bookmarks/\\" class=\\"menu-item\\">
			      Bookmarks Manager
			    </a>
			    <a href=\\"chrome://downloads/\\" class=\\"menu-item\\">
			      Downloads
			    </a>
			    <a href=\\"chrome://history/\\" class=\\"menu-item\\">
			      History
			    </a>
			    <a href=\\"chrome://settings/passwords\\" class=\\"menu-item\\">
			      Passwords
			    </a>

			    <hr>

			    <a id=\\"o\\" class=\\"menu-item\\">
			      New Tab Settings
			    </a>
			    <a href=\\"https://github.com/MaxMilton/new-tab/issues\\" class=\\"menu-item\\">
			      Submit Bug
			    </a></div></div>"
`;
